examples/resources/xray_watch/resource.tf

```diff
@@ -0,0 +1,87 @@
+resource "xray_watch" "all-repos" {
+  name        = "all-repos-watch"
+  description = "Watch for all repositories, matching the filter"
+  active      = true
+  watch_resource {
+    type = "all-repos"
+    filter {
+      type  = "regex"
+      value = ".*"
+    }
+  }
+  assigned_policy {
+    name = xray_security_policy.allowed_licenses.name
+    type = "security"
+  }
+  assigned_policy {
+    name = xray_license_policy.banned_licenses.name
+    type = "license"
+  }
+  watch_recipients = ["test@email.com", "test1@email.com"]
+}
+
+resource "xray_watch" "repository" {
+  name        = "repository-watch"
+  description = "Watch a single repo or a list of repositories"
+  active      = true
+
+  watch_resource {
+    type       = "repository"
+    bin_mgr_id = "default"
+    name       = "your-repository-name"
+    filter {
+      type  = "regex"
+      value = ".*"
+    }
+  }
+
+  watch_resource {
+    type       = "repository"
+    bin_mgr_id = "default"
+    name       = "your-other-repository-name"
+    filter {
+      type  = "package-type"
+      value = "Docker"
```

Comment:
Do we actually validate that it's one of our supported package types?

Reply:
no, we don't verify the value in the filters. The type could be in `"regex", "package-type", "ant-patterns"`
Each type should be verified separately. Not sure if we can validate the value depending on the other value in the schema.